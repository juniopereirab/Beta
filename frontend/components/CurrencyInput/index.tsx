import React from 'react'
import { formatCurrencyNumber } from '../../utils/formaters'
import { TextField } from '@mui/material'

interface ICurrencyInput {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value: number
  name: string
  id: string
  label: string

}

const CurrencyInput: React.FC<ICurrencyInput> = ({ onChange, value, name, id, label }) => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target

    const floatValue = parseFloat(value.replace(/[^\d]/g, ''))

    if (Number.isNaN(floatValue)) {
      const newEvent = event
      newEvent.target.value = '0'
      return onChange(newEvent)
    }

    const newEvent = event
    newEvent.target.value = String(floatValue / 100)
    onChange(newEvent)
  }

  return (
    <TextField
      value={formatCurrencyNumber(value)}
      onChange={handleChange}
      fullWidth
      name={name}
      id={id}
      label={label}
    />
  )
}

export default CurrencyInput