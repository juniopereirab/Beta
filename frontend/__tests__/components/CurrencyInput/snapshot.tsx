import { render } from '@testing-library/react'
import CurrencyInput from '@/components/CurrencyInput'
 
it('renders input correctly', () => {
  const { container } = render(<CurrencyInput id='test' label='Teste' name='test' onChange={() => {}} value={0}/>)
  expect(container).toMatchSnapshot()
})