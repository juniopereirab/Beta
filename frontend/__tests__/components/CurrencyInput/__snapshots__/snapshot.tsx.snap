// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="test"
      id="test-label"
    >
      Teste
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="test"
        name="test"
        type="text"
        value="R$ 0,00"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Teste
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`renders input correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="test"
      id="test-label"
    >
      Teste
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="test"
        name="test"
        type="text"
        value="R$ 0,00"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Teste
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
